File: AtCoder/ABC/C/249/C.cpp
#include<iostream>
#include<vector>
#include<map>
#include<algorithm>
using namespace std;

int main(){
    // INPUT
    int n, k, max_cnt = 0;
    cin >> n >> k;
    vector<string> ss(n);
    for(int i=0; i<n; i++) cin >> ss.at(i);

    for(int bit=0; bit<(1<<n); bit++){
        map<int, int> mp;
        for(int i=0; i<n; i++){
            if(bit & (1<<i)){
                for(auto c: ss.at(i)){
                    mp[c] += 1;
                }
            }
        }
        int cnt = 0;
        for(auto iter: mp){
            if(iter.second == k) cnt++;
        }   
        max_cnt = max(max_cnt, cnt);
    }
    cout << max_cnt << endl;
}

File: AtCoder/ABC/C/249/C.rb
n, k = gets.chomp.split.map(&:to_i)
ss = Array.new
n.times do
    ss << gets.chomp
end
max_cnt = 0
n.times do |i|
    ss.combination(i+1).each do |comb|
        hash = Hash.new(0)
        comb.to_a.each do |s|
            s.chars.each do |c|
                hash[c] += 1
            end
        end
        cnt = 0
        hash.each do |key, val|
            cnt += 1 if val == k
        end
        max_cnt = [max_cnt, cnt].max
    end
end
p max_cnt